app/src/test/java/waiter/MessengerTest.java

```diff
@@ -5,17 +5,40 @@
 import waiter.Messenger.Messenger;
 
 import org.junit.jupiter.api.Test;
-import static org.junit.jupiter.api.Assertions.assertEquals;
+import org.junit.jupiter.api.BeforeEach;
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
 
 import java.io.IOException;
+import java.util.Arrays;
 
 class MessengerTest {
+
+    String clientHasDisconnected;
+
+    @BeforeEach
+    void setUp() {
+        clientHasDisconnected = null;
+    }
+
+    @Test
+    void echosSingleMessage() throws IOException {
+        String[] clientInputs = {"foo", clientHasDisconnected};
+        String[] expected = Arrays.copyOfRange(clientInputs, 0, clientInputs.length - 1);
+        ClientConnectionMock clientConnectionMock = new ClientConnectionMock(clientInputs);
+        EchoProtocolMock echoProtocolMock = new EchoProtocolMock();
+        Messenger messenger = new Messenger(echoProtocolMock);
+        messenger.transport(clientConnectionMock);
+        assertArrayEquals(expected, clientConnectionMock.echoedInputs);
```

Comment:
Maybe it's easier if we just do `assertArrayEquals({"foo"}, clientConnectionMock.echoedInputs);` and we don't need to use `Arrays.copyOfRange` up there.

Reply:
We can do that, I thought the assert might be harder to read that way, but the more I think about it, although the assert is easier to read, the expected assignment looks like a bit of a mess. Ill make this change now, thank you! 😄

Reply:
Addressed in [this commit](https://github.com/jonmanahan/waiter/pull/7/commits/97d783bb1cde52ca12d9dc231d05bf967e161ca5), fixed the test name, some more test format and added a comment to the null variable declaration.